lua/lualine.lua

```diff
@@ -148,34 +150,66 @@ end
 
 local function statusline(sections, is_focused)
   if M.theme ~= theme_set then
-    set_lualine_theme()
-  end
-  local status = {}
-  if sections.lualine_a then
-    local hl = highlight.format_highlight(is_focused, 'lualine_a')
-    table.insert(status, utils_component.draw_section(sections.lualine_a,  hl))
-  end
-  if sections.lualine_b then
-    local hl = highlight.format_highlight(is_focused, 'lualine_b')
-    table.insert(status, utils_component.draw_section(sections.lualine_b,  hl))
-  end
-  if sections.lualine_c then
-    local hl = highlight.format_highlight(is_focused, 'lualine_c')
-    table.insert(status, utils_component.draw_section(sections.lualine_c,  hl))
+    _G.lualine_set_theme()
   end
-  table.insert(status, "%=")
-  if sections.lualine_x then
-    local hl = highlight.format_highlight(is_focused, 'lualine_c')
-    table.insert(status, utils_component.draw_section(sections.lualine_x,  hl))
-  end
-  if sections.lualine_y then
-    local hl = highlight.format_highlight(is_focused, 'lualine_b')
-    table.insert(status, utils_component.draw_section(sections.lualine_y,  hl))
+  -- status_builder stores statusline without section_separators
+  local status_builder = {}
+  -- The sequence sections should maintain
+  local section_sequence = {'a', 'b', 'c', 'x', 'y', 'z'}
+
+  for _, section_name in ipairs(section_sequence) do
+    if sections['lualine_'..section_name] then
+      -- insert highlight+components of this section to status_builder
+      local section_highlight = highlight.format_highlight(is_focused,
+                   'lualine_'..section_name)
+			local section_data = utils_component.draw_section(sections['lualine_'..section_name], section_highlight)
+			if #section_data > 0 then
+				table.insert(status_builder, {section_name, section_data})
```

Comment:
I mentioned this before. It would be way better to have `status_builder` as a map
`{name = section_name, data=section_data}`
this will make it a lot easier to read down. instead of having `[1]` `[2]` everywhere

Reply:
I first tried map But maps doesn't store sequencial data . When cection exists and which setion comes after what is nessary for this to work

Reply:
I'm talking about an array of maps.... Instead of `status_builder[i][2]` we can have `status_builder[i].data`. way cleaner

Reply:
Ok that's posible

Reply:
applied

Reply:
nice! looks way better now